shared/src/main/java/uk/gov/di/authentication/shared/entity/NotificationType.java

```diff
@@ -1,21 +1,60 @@
 package uk.gov.di.authentication.shared.entity;
 
+import uk.gov.di.authentication.shared.helpers.LocaleHelper.SupportedLanguage;
+import uk.gov.di.authentication.shared.services.ConfigurationService;
+
+import java.util.HashMap;
+import java.util.Map;
+
 public enum NotificationType implements TemplateAware {
-    VERIFY_EMAIL("VERIFY_EMAIL_TEMPLATE_ID"),
-    VERIFY_PHONE_NUMBER("VERIFY_PHONE_NUMBER_TEMPLATE_ID"),
-    MFA_SMS("MFA_SMS_TEMPLATE_ID"),
-    RESET_PASSWORD("RESET_PASSWORD_TEMPLATE_ID"),
-    PASSWORD_RESET_CONFIRMATION("PASSWORD_RESET_CONFIRMATION_TEMPLATE_ID"),
-    ACCOUNT_CREATED_CONFIRMATION("ACCOUNT_CREATED_CONFIRMATION_TEMPLATE_ID"),
-    RESET_PASSWORD_WITH_CODE("RESET_PASSWORD_WITH_CODE_TEMPLATE_ID");
+    VERIFY_EMAIL(
+            "VERIFY_EMAIL_TEMPLATE_ID",
+            Map.of(SupportedLanguage.CY, "VERIFY_EMAIL_TEMPLATE_ID_CY")),
+    VERIFY_PHONE_NUMBER(
+            "VERIFY_PHONE_NUMBER_TEMPLATE_ID",
+            Map.of(SupportedLanguage.CY, "VERIFY_PHONE_NUMBER_TEMPLATE_ID_CY")),
+    MFA_SMS("MFA_SMS_TEMPLATE_ID", Map.of(SupportedLanguage.CY, "MFA_SMS_TEMPLATE_ID_CY")),
+    RESET_PASSWORD(
+            "RESET_PASSWORD_TEMPLATE_ID",
+            Map.of(SupportedLanguage.CY, "RESET_PASSWORD_TEMPLATE_ID_CY")),
+    PASSWORD_RESET_CONFIRMATION(
+            "PASSWORD_RESET_CONFIRMATION_TEMPLATE_ID",
+            Map.of(SupportedLanguage.CY, "PASSWORD_RESET_CONFIRMATION_TEMPLATE_ID_CY")),
+    ACCOUNT_CREATED_CONFIRMATION(
+            "ACCOUNT_CREATED_CONFIRMATION_TEMPLATE_ID",
+            Map.of(SupportedLanguage.CY, "ACCOUNT_CREATED_CONFIRMATION_TEMPLATE_ID_CY")),
+    RESET_PASSWORD_WITH_CODE(
+            "RESET_PASSWORD_WITH_CODE_TEMPLATE_ID",
+            Map.of(SupportedLanguage.CY, "RESET_PASSWORD_WITH_CODE_TEMPLATE_ID_CY"));
 
     private final String templateName;
 
-    NotificationType(String templateName) {
+    private static final ConfigurationService configurationService = new ConfigurationService();
+    private Map<SupportedLanguage, String> languageSpecificTemplates = new HashMap<>();
+
+    private NotificationType(String templateName) {
         this.templateName = templateName;
     }
 
+    private NotificationType(
+            String templateName, Map<SupportedLanguage, String> languageSpecificTemplates) {
+        this(templateName);
+        this.languageSpecificTemplates = languageSpecificTemplates;
+    }
+
     public String getTemplateId() {
         return System.getenv(templateName);
     }
+
+    public String getTemplateId(SupportedLanguage language) {
+        return configurationService.getNotifyTemplateId(getTemplateName(language));
```

Comment:
Do we want to return the ID for the English version of the template if, for whatever reason, the language specific template ID can't be found? Belt and braces!

Reply:
Yes good call, it does actually do that at the moment if you look at NotificationType:

   String getTemplateName(SupportedLanguage language) {
        if (languageSpecificTemplates.containsKey(language)) {
            return languageSpecificTemplates.get(language);
        } else {
            return templateName;
        }
    }

If it doesn't find the language in languageSpecificTemplates (which contains Welsh only) then it goes back to the default.

Until the next PR this won't actually be in use, it's currently still using the existing getTemplateId()